pkg/plugin/driver/daemonset/daemonset_test.go

```diff
@@ -166,3 +171,153 @@ func TestFillTemplate(t *testing.T) {
 		t.Errorf("Expected annotations key1:val1 and key2:val2 to be set, but got %v", daemonSet.Spec.Template.Annotations)
 	}
 }
+
+func TestMonitorOnce(t *testing.T) {
+	// Note: the pods/ds must be marked with the label "sonobuoy-run" or else our labelSelector
+	// logic will filter them out even though the fake server returns them.
+
+	// We will need to be able to grab these items repeatedly and tweak the minor details
+	// so these helpers make the test cases much more readable.
+	testPlugin := &Plugin{Base: driver.Base{Definition: plugin.Definition{Name: "myPlugin"}}}
+	validDS := appsv1.DaemonSet{
+		ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"sonobuoy-run": ""}},
+	}
+	validPod := func(node string) corev1.Pod {
+		return corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"sonobuoy-run": ""}},
+			Spec:       corev1.PodSpec{NodeName: node},
+		}
+	}
+	failingPod := func(node string) corev1.Pod {
+		return corev1.Pod{
+			ObjectMeta: metav1.ObjectMeta{Labels: map[string]string{"sonobuoy-run": ""}},
+			Spec:       corev1.PodSpec{NodeName: node},
+			Status: corev1.PodStatus{
+				Conditions: []corev1.PodCondition{
+					{Reason: "Unschedulable", Message: "conditionMsg"},
+				},
+			},
+		}
+	}
+	default3Nodes := []corev1.Node{
+		{ObjectMeta: metav1.ObjectMeta{Name: "node1"}},
+		{ObjectMeta: metav1.ObjectMeta{Name: "node2"}},
+		{ObjectMeta: metav1.ObjectMeta{Name: "node3"}},
+	}
+
+	testCases := []struct {
+		desc       string
+		expectDone bool
+
+		// Ensure we are getting the err result we expect; lots of ways to get errors
+		// that may not be clear.
+		expectErrResultMsgs []string
+
+		dsPlugin                *Plugin
+		dsOnServer              *appsv1.DaemonSetList
+		podsOnServer            *corev1.PodList
+		nodes                   []corev1.Node
+		errFromServerForDSList  error
+		errFromServerForPodList error
+	}{
+		{
+			desc:       "Cleaned up indicates exit without error",
+			expectDone: true,
+			dsPlugin:   &Plugin{driver.Base{CleanedUp: true}},
+		}, {
+			desc:       "Missing daemonset results in no errors",
+			dsPlugin:   testPlugin,
+			dsOnServer: &appsv1.DaemonSetList{},
+		}, {
+			desc:                    "Failed pod lookup results in no error",
+			dsPlugin:                testPlugin,
+			dsOnServer:              &appsv1.DaemonSetList{Items: []appsv1.DaemonSet{validDS}},
+			podsOnServer:            &corev1.PodList{},
+			errFromServerForPodList: errors.New("pod lookup err"),
+		}, {
+			desc:         "Missing pods results in errors for each",
+			nodes:        default3Nodes,
+			dsPlugin:     testPlugin,
+			dsOnServer:   &appsv1.DaemonSetList{Items: []appsv1.DaemonSet{validDS}},
+			podsOnServer: &corev1.PodList{},
+			expectErrResultMsgs: []string{
+				"No pod was scheduled on node node1 within 2562047h47m16.854775807s. Check tolerations for plugin myPlugin",
```

Comment:
I'm guessing we have this specific time value because we don't have a creation timestamp for the DaemonSet and this is the string representation of the maximum duration. Perhaps we could add a comment for this, or check only a portion of the error message. e.g. check that it contains `No pod was scheduled on node node1 within`

Reply:
Checking substrings IMO is slightly awkward and this also makes it clear in the tests what the whole message we surface to the user includes (which is what informed your idea in your other comment to add info to the error message).

If this is just a nit I'm inclined to just leave this as is since it is just a minor testing quirk which I don't think would confuse a dev too bad.

Reply:
Fair enough. I agree that it's not very confusing, but I did have to think about why this expected message included the duration value that it did. I think I just tend to be overly cautious of tests related to time calculations :) In this case, I wanted to be sure that the test wouldn't incorrectly fail at some point in the future because of a different duration being included in the message. I can include adding a comment here as part of the follow up issue that I'll make for the other tests.